src/device-registry/utils/date.js

```diff
@@ -1,5 +1,12 @@
 const { logText, logObject, logElement } = require("./log");
 const isEmpty = require("is-empty");
+
+
+module.exports={
```

Comment:
Hi @BenjaminSsempala , isn't this export already done at the bottom of this file?

Could this be a repetition?  Please see attached screenshot.
<img width="538" alt="Screenshot 2022-09-30 at 18 16 08" src="https://user-images.githubusercontent.com/1590213/193302229-c4da1a58-57c5-4729-a81e-18a566d4acff.png">

Reply:
I guess we just need to dig deeper into this. Quite an interesting challenge we have. And it also helps us learn more about the language (Javascript)! :)

This could also be an opportunity for us to utilise VS Code debugging tools available where we can step over code and observe the flow of data in real time.

Thanks @BenjaminSsempala

Reply:
Alright @BenjaminSsempala , please add it back then since I had removed it in my commits. I was just not sure of its relevance.

For the start, please git pull all my latest changes and test out before making any more changes. Thanks!

My commits from yesterday just modified the entire function so that some of the calculations in the constants.js are done from outside and their results shared as function inputs. This seemed to eradicate the issue from my local development tests.

Looking forward to your findings.

Reply:
Thank you @Baalmart .
I've run this locally and still get a circular dependency warning. 
![image](https://user-images.githubusercontent.com/86492979/193673181-e54468da-4cb7-4798-81c7-7213c4cc8421.png)
However when I import the `generateDateFormatWithoutHrs` function, It seems to be running well.
![image](https://user-images.githubusercontent.com/86492979/193681384-65f399ed-7818-4c31-a69d-988e1351f917.png)